pkg/virtualKubelet/reflection/workload/podns.go

```diff
@@ -718,3 +719,28 @@ func (npr *NamespacedPodReflector) InferAdditionalRestarts(local, remote *corev1
 	// This should never occur, since the containers should match
 	return 0
 }
+
+// List retrieves the list of reflected pods.
+func (npr *NamespacedPodReflector) List() ([]interface{}, error) {
+	var list []interface{}
+
+	objs, err := npr.localPods.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	tmplist := make([]interface{}, len(objs))
+	for i := range objs {
+		tmplist[i] = virtualkubelet.ForgeNamespacedName(objs[i])
+	}
+
+	objs, err = npr.remotePods.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	tmplist = make([]interface{}, len(objs))
+	for i := range objs {
+		tmplist[i] = virtualkubelet.ForgeNamespacedName(objs[i])
+	}
+
+	return list, nil
```

Comment:
You are probably missing some copies from/to tmpLists (in both cases to/from the list returned

additionally, you don't need the tmpList at all